community/kernel/src/main/java/org/neo4j/kernel/impl/api/KernelTransactionImplementation.java

```diff
@@ -215,6 +221,8 @@ public KernelTransactionImplementation initialize(
         this.transactionEvent = tracer.beginTransaction();
         assert transactionEvent != null : "transactionEvent was null!";
         this.accessMode = accessMode;
+        this.transactionId = NOT_COMMITTED_TRANSACTION_ID;
+        this.commitTime = NOT_COMMITTED_TRANSACTION_COMMIT_TIME;
```

Comment:
Since we pool transactions, isn't there a race between reading these fields after the transaction is closed and the re-usage of the object?